core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -5682,11 +5682,21 @@ public static List mapEntries(Map<Object, Object> map) {
 
   /** Support the MAP_KEYS function. */
   public static List mapKeys(Map map) {
+    for (Object key : map.keySet()) {
+      if (key == null) {
+        throw new IllegalArgumentException("Cannot use null as map key");
```

Comment:
You should `throw RESOURCE.something`. We have resources for error messages that the user might see during normal operation.

Reply:
done